src/apps/HeyGPT/HeyGPT/Program.cs

```diff
@@ -0,0 +1,111 @@
+﻿// If you need to attach the debugger early when you are not running in the IDE then
+// you can uncomment the #define WAIT_FOR_DEBUGGER_TO_ATTACH. This will display a message box
+// pausing execution of the application allowing you to attach the debugger.  Dismiss the message
+// box and program execution will continue.
+
+// #define WAIT_FOR_DEBUGGER_TO_ATTACH
+
+#nullable disable
+
+using Microsoft.UI.Dispatching;
+using Microsoft.Windows.AppLifecycle;
+using WinRT;
+using System.Diagnostics;
+using HeyGPT.App.Helpers;
+
+namespace HeyGPT.App;
+
+public class Program
+{
+    private const string AppKey = "HeyGPT_3AFC048E-1B59-48D4-8DC0-A98E952482F0";
+
+    // We override the default Main function to enforce single instance mechanisms.
+    // We need single instance support to allow us to handle URI protocols needed
+    // for communication with other process such as web browsers when doing OAuth
+    // redirection.  It also allows us to trigger the application to the foreground
+    // when the user tries to run the application a second time.
+
+    [STAThread]
+    private static async Task<int> Main(string[] args)
+    {
+        // See above
+        DebuggerHelper.WaitForDebuggerToAttach();
+
+        ComWrappersSupport.InitializeComWrappers();
+
+        var isRedirect = await DecideRedirection().ConfigureAwait(false);
+
+        if (!isRedirect)
+        {
+            // This is the first instance of the application, so we start the application
+            // object.
+
+            Microsoft.UI.Xaml.Application.Start((p) =>
+            {
+                var context = new DispatcherQueueSynchronizationContext(
+                    DispatcherQueue.GetForCurrentThread());
+                SynchronizationContext.SetSynchronizationContext(context);
+                new App();
+            });
+
+            // The application has been closed and is about the exit.
+            // We clean up here...
+
+            AppInstance.GetCurrent().UnregisterKey();
+        }
+
+        return 0;
+    }
+
+    private static async Task<bool> DecideRedirection()
+    {
+        var isRedirect = false;
+
+        try
+        {
+            var appActivationArguments = AppInstance.GetCurrent().GetActivatedEventArgs();
+            var extendedActivationKind = appActivationArguments.Kind;
+
+            var appInstance = AppInstance.FindOrRegisterForKey(AppKey);
+
+            // Handle any activation scenario that should be processed before the
+            // application starts.
+            HandleActivation(appActivationArguments);
```

Comment:
There may be reasons to handle the activation code here as well... Perhaps you want to apply updates or perform a utility without bringing up the UX.   You can do all that here before the App object is created.